patches/minecraft/net/minecraft/world/entity/monster/Zombie.java.patch

```diff
@@ -1,5 +1,18 @@
 --- a/net/minecraft/world/entity/monster/Zombie.java
 +++ b/net/minecraft/world/entity/monster/Zombie.java
+@@ -156,11 +_,7 @@
+    }
+ 
+    protected int m_6552_(Player p_34322_) {
+-      if (this.m_6162_()) {
+-         this.f_21364_ = (int)((float)this.f_21364_ * 2.5F);
+-      }
+-
+-      return super.m_6552_(p_34322_);
++      return this.m_6162_() ? (int) (super.m_6552_(p_34322_) * 2.5F) : super.m_6552_(p_34322_);
```

Comment:
This logic is incompatible with vanilla. Looking at `Mob#getExperienceReward`, the experience is first cached than added onto based on armor and weapons. This means that the baby multiplier would then be apply 2.5x the amount of the total experience gained instead of just the zombie experience.

Reply:
I have changed this to now use a temporary variable to restore the xpReward field after the super method is called. I am not happy with this solution, but to me it is the most clear as to what is happening, while preserving the currently intended vanilla mechanic.

```java
protected int getExperienceReward(Player p_34322_) {
      if (this.isBaby()) {
         int oldReward = this.xpReward;
         this.xpReward = (int)((float)this.xpReward * 2.5F);
         int reward = super.getExperienceReward(p_34322_);
         this.xpReward = oldReward;
         return reward;
      }

      return super.getExperienceReward(p_34322_);
   }
```

If we are unhappy with this temp variable assigment, an equivalent EXP calculation can be made by recognizing that 1x of the xpReward is accounted for in the super call, and we can just add the difference back after the calculation without assignment.

```java
protected int getExperienceReward(Player p_34322_) {
      if (this.isBaby()) {
         return super.getExperienceReward(p_34322_) + (int) this.xpReward * 1.5F;
      }

      return super.getExperienceReward(p_34322_);
   }
```